pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -77,28 +77,29 @@ type kubenetNetworkPlugin struct {
 	hairpinMode     componentconfig.HairpinMode
 	hostPortMap     map[hostport]closeable
 	iptables        utiliptables.Interface
+	cniPath         string
 }
 
-func NewPlugin() network.NetworkPlugin {
+func NewPlugin(cniPath string) network.NetworkPlugin {
 	protocol := utiliptables.ProtocolIpv4
 	execer := utilexec.New()
 	dbus := utildbus.New()
 	iptInterface := utiliptables.New(execer, dbus, protocol)
-
 	return &kubenetNetworkPlugin{
 		podIPs:      make(map[kubecontainer.ContainerID]string),
 		hostPortMap: make(map[hostport]closeable),
 		MTU:         1460, //TODO: don't hardcode this
 		execer:      utilexec.New(),
 		iptables:    iptInterface,
+		cniPath:     cniPath,
 	}
 }
 
 func (plugin *kubenetNetworkPlugin) Init(host network.Host, hairpinMode componentconfig.HairpinMode) error {
 	plugin.host = host
 	plugin.hairpinMode = hairpinMode
 	plugin.cniConfig = &libcni.CNIConfig{
-		Path: []string{DefaultCNIDir},
+		Path: []string{DefaultCNIDir, plugin.cniPath},
```

Comment:
CNI search for plugins sequentially. Maybe I should put the cniPath before DefaultCNIDir?